### PR TITLE
feat(Authentication): Add password security policy

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds a new page to configure the password security policy.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Test the page thoroughly based on the specifications from the ticket MON-6463

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by kduret** [comment]
waiting backend storage refacto before ready to review

**Review by AssanDiomande — DISMISSED** [review]

**Review by Jordan91 — DISMISSED** [review]

File: www/front_src/src/Authentication/Form/PasswordBlockingPolicy/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      screen.getByLabelText(labelNumberOfAttemptsBeforeBlockingNewAttempts),
+      '11',
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(labelChooseAValueBetween1and10),
+      ).toBeInTheDocument();
+    });
+  });
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordBlockingPolicy/index.test.tsx:121]: I would merge these two cases into one to optimize the tests runtime. We can use a name like: "it displays an error message when the number of attemps is outside the bounds"

File: www/front_src/src/Authentication/Form/PasswordBlockingPolicy/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+  it('displays an error message in the "Time blocking duration" field when the number of attempts is emptied', async () => {
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordBlockingPolicy/index.test.tsx:123]:
```suggestion
  it('displays an error message in the "Time blocking duration" field when the number of attempts is cleared', async () => {
```

File: www/front_src/src/Authentication/Form/PasswordBlockingPolicy/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    });
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(labelBlockingDurationMustBeLessThanOrEqualTo7Days),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it('displays "Strong" when the number of attempts is 2', async () => {
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordBlockingPolicy/index.test.tsx:160]: Same remark as above, we can group the "Strong" and "Good" cases

File: www/front_src/src/Authentication/Form/PasswordCasePolicy/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+    userEvent.type(
+      screen.getByLabelText(labelPasswordLength),
+      '{selectall}{backspace}45',
+    );
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(labelPasswordLength)).toHaveValue(45);
+    });
+  });
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordCasePolicy/index.test.tsx:96]:
```suggestion
  });
  
```

File: www/front_src/src/Authentication/Form/PasswordCasePolicy/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  it('displays an error message when password minimum length input is emptied', async () => {
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordCasePolicy/index.test.tsx:97]:
```suggestion
  it('displays an error message when password minimum length input is cleared', async () => {
```

File: www/front_src/src/Authentication/Form/PasswordCasePolicy/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    expect(
+      screen.getAllByLabelText(labelForceToUseNumbers)[0],
+    ).toBeInTheDocument();
+    expect(
+      screen.getAllByLabelText(labelForceToUseSpecialCharacters)[0],
+    ).toBeInTheDocument();
+    expect(screen.getByText(labelStrong)).toBeInTheDocument();
+  });
+
+  it('changes the password minimum length value when "45" is typed in the input', async () => {
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordCasePolicy/index.test.tsx:85]: Same remark here regarding the test cases grouping

File: www/front_src/src/Authentication/Form/PasswordExpirationPolicy/index.test.tsx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    expect(screen.getByText(labelPasswordExpiration)).toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(labelChooseADurationBetween7DaysAnd12Months),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it('does not display any error message when the password expiration time is emptied', async () => {
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordExpirationPolicy/index.test.tsx:113]:
```suggestion
  it('does not display any error message when the password expiration time is cleared', async () => {
```

File: www/front_src/src/Authentication/Form/PasswordExpirationPolicy/index.test.tsx [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    ).toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(labelChooseADurationBetween1HourAnd1Week),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it('does not display any error message when the delay before new password time is emptied', async () => {
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordExpirationPolicy/index.test.tsx:155]:
```suggestion
  it('does not display any error message when the delay before new password time is cleared', async () => {
```

**Review by bdauria — DISMISSED** [review]

**Review by jdelpierre — DISMISSED** [review]
OK FE Part

**Review by AssanDiomande — DISMISSED** [review]

**Pull request merged**